gui/default/syncthing/core/syncthingController.js

```diff
@@ -1226,6 +1227,9 @@ angular.module('syncthing.core')
 
         $scope.saveDevice = function () {
             $('#editDevice').modal('hide');
+            if ($scope.currentDevice.deviceID === '') {
+                $scope.currentDevice.deviceID = $scope.manualDeviceID;
+            }
```

Comment:
I wonder if something like this works instead, removing the need for a scope variable

```
         <select ng-if="!editingExisting && discovery" class="form-control" ng-model="currentDevice.deviceID">
            <option ng-repeat="(id, data) in discovery" value="{{id}}">{{id}}</option>
            <option ng-value='currentDevice.deviceID' translate>Enter manually below</option>
          </select>
```